_posts/2017-12-01-CocoaPods-1.4.0.markdown

```diff
@@ -0,0 +1,201 @@
+
+  s.private_header_files = 'Sources/Internal/**/*.h'
+end
+```
+
+The above will automatically add a script phase to the `CannonPodder-iOS` and `CannonPodder-macOS` targets generated by CocoaPods. It looks like this:
+
+<center>
+{% breaking_image /assets/blog_img/CocoaPods-1.4.0/script_phases_podspec_1.png %}
+</a></center>
+
+If your pod supports multiple platforms, you can limit a script phase only to a specific platform, for example:
+
+```ruby
+s.ios.script_phase = { :name => 'Hello iOS World', :script => 'echo "Hello iOS World"' }
+```
+
+This will only add a script phase to the `CannonPodder-iOS` target.
+
+Additionally, there are times where you would like your script phases to be executed *before* compilation in order to perform a configuration step or a check. Support for this is included with CocoaPods 1.4.0 via the `:execution_position` attribute within each `script_phase` entry.
+
+```ruby
+s.script_phase = { 
+  :name => 'Hello World', 
+  :script => 'echo "Hello World"', 
+  :execution_position => :before_compile 
+}
+```
+
+Which generates the following:
+
+<center>
+{% breaking_image /assets/blog_img/CocoaPods-1.4.0/script_phases_podspec_2.png %}
+</a></center>
+
+CocoaPods will automatically add the script phase to be executed before the `Compile Sources` phase executes. This effectively allows you to control at what stage you prefer a script phase to be executed.
+
+Here's a slightly more complicated example that demonstrates two script phases running at different stages:
+
+```ruby
+s.script_phases = [
+  { :name => 'Precompile', :script => './setup.sh', :execution_position => :before_compile },
+  { :name => 'Postcompile', :script => 'echo "yay!"', :execution_position => :after_compile }
+]
+```
+
+Currently, the available execution positions are `:before_compile`, `:after_compile` and `:any` which is the default.
+
+**Note**: If you'd like your script phase to access any of the sources of your pod then use the `${PODS_TARGET_SRCROOT}` environment variable which is set for each pod and is always relative to the path of your pod sources.
+
+#### Security Warning
+
+While script phases are a powerful addition to CocoaPods, it is crucial to understand that consuming pods that include script phases may be a high security vulnerability for many. After all, this feature essentially allows pod authors to execute code on your machine if you decide to integrate their pod into your project.
+
+For this reason, CocoaPods will **always** display a warning when you install a pod that includes script phases.
+
+A warning would be displayed as such (click to enlarge):
+
+<center>
+{% breaking_image /assets/blog_img/CocoaPods-1.4.0/script_phases_security_1.png %}
+</a></center>
+
+**Note**: This warning cannot be disabled and it will only appear the first time you install a pod. It is imperative that you inspect the contents of the script phase to ensure no harmful code is executed as part of your compilation process.
```

Comment:
Does the "first time you install a pod" include upgrading to a new version? In that case, would want to be reminded to check the script in case it has changed

Reply:
good point. I should update this text to reflect this. thank you.